specifyweb/frontend/js_src/lib/components/Statistics/hooks.tsx

```diff
@@ -250,42 +208,36 @@ export function useResolvedStatSpec(
   }, [item]);
 }
 
-export function setAbsentCategoriesToFetch(
-  layout: StatLayout | undefined,
-  currentCategoriesToFetch: RA<string>,
-  setCategoriesToFetch: (currentCategories: RA<string>) => void
-) {
-  if (layout === undefined) return;
-  const categoriesToFetch: WritableArray<string> = [];
-  layout.forEach((pageLayout) =>
-    pageLayout.categories.forEach(({ items, categoryToFetch }) => {
-      if (items === undefined && categoryToFetch !== undefined) {
-        categoriesToFetch.push(categoryToFetch);
-      }
-      (items ?? []).forEach((item) => {
-        if (
-          item.type === 'DefaultStat' &&
-          item.itemType === 'BackEndStat' &&
-          item.itemValue === undefined &&
-          item.itemName === 'phantomItem'
+export function getDynamicCategoriesToFetch(
+  layout: RA<StatLayout> | undefined
```

Comment:
Having a function just accept one argument that is undefined is a bit of a code smell - just don't call the function at all.
Another way to think of it is that a function is for taking input, processing it and returning it - if you don't have input to give, then you have no business calling the function.

Instead, before calling the function, check for undefined (or use `f.maybe`). An exception to this rule is when the function is called with undefined in a lot of places (which is not the case here)